pac4j-saml/src/main/java/org/pac4j/saml/store/HttpSessionStore.java

```diff
@@ -86,17 +89,27 @@ public void set(final String messageID, final XMLObject message) {
      */
     @Override
     public Optional<XMLObject> get(final String messageID) {
-        final LinkedHashMap<String, XMLObject> messages = getMessages();
-        final XMLObject o = messages.get(messageID);
+        final LinkedHashMap<String, String> messages = getMessages();
+        final String o = messages.get(messageID);
         if (o == null) {
-            log.debug("Message {} not found in session {}", messageID, context.getSessionStore().getOrCreateSessionId(context));
+            log.debug("Message {} not found in session {}",
+                    messageID, context.getSessionStore().getOrCreateSessionId(context));
```

Comment:
Not sure you need to log `context.getSessionStore().getOrCreateSessionId(context)`, as that might actually create a session id? @leleuj would know best.

Reply:
I left the log statement as it was.
Please let me know if you prefer to change that line

Reply:
Indeed, the `getOrCreateSessionId` method could create a session if it does not exist.
Given it's in the webflow of the SAML login process, it's unlikely though.
I think we can keep these for now.

Reply:
thanks